File: dp/minimum_operations.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends

class Solution {
 public:
  int minOperation(int n) {
    // code here.
    vector<int> dp(n + 1);
    dp[1] = 1;
    for (int i = 2; i <= n; ++i) {
      dp[i] = dp[i - 1] + 1;
      if (i % 2 == 0) {
        dp[i] = min(dp[i], dp[i / 2] + 1);
      }
    }
    return dp[n];
  }
};

//{ Driver Code Starts.
int main() {
  int t;
  cin >> t;
  while (t--) {
    int n;
    cin >> n;
    Solution ob;
    cout << ob.minOperation(n) << endl;
  }
}
// } Driver Code Ends

File: math/minimum_operations.py
# User function Template for python3


class Solution:
    def minOperation(self, n):
        # code here
        ans = 0
        while n > 0:
            if n % 2 == 1:
                n -= 1
            else:
                n >>= 1
            ans += 1
        return ans


# {
# Driver Code Starts
# Initial Template for Python 3

if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        n = int(input())
        ob = Solution()
        print(ob.minOperation(n))
# } Driver Code Ends
